File: data/code_pairs/fortran/544.f90
module kpoints
    
    implicit none

    ! Declarations
    integer nkpoints
    real, dimension (:, :), allocatable :: special_k
    real, dimension (:, :), allocatable :: special_k_orig
    real, dimension (:), allocatable :: weight_k
    real, dimension (:), allocatable :: weight_k_orig
    real, dimension (:, :), allocatable :: scale_k
    character (len = 40) kptpreference 

end module kpoints

File: data/code_pairs/cpp/544.cpp
#include <vector>
#include <string>

class KPoints {
public:
    int nkpoints;
    std::vector<std::vector<float>> special_k;
    std::vector<std::vector<float>> special_k_orig;
    std::vector<float> weight_k;
    std::vector<float> weight_k_orig;
    std::vector<std::vector<float>> scale_k;

    std::string kptpreference;

    void allocateSpecialK(int dim1, int dim2) {
        special_k.resize(dim1, std::vector<float>(dim2));
    }

    // Optionally, other member functions...
};